core/src/main/scala/com/clairvoyant/restonomer/core/app/RestonomerContext.scala

```diff
@@ -30,21 +30,21 @@ class RestonomerContext(val restonomerContextDirectoryPath: String) {
 
   val configVariables: Map[String, String] = loadConfigVariables(CONFIG_VARIABLES_FILE_PATH)
 
-  val checkpoints: List[CheckpointConfig] = loadConfigsFromDirectory[CheckpointConfig](
-    configDirectoryPath = CHECKPOINTS_CONFIG_DIRECTORY_PATH,
-    configVariablesSubstitutor = ConfigVariablesSubstitutor(configVariables = configVariables)
-  )
-
-  def runCheckpoint(checkpointName: String): Unit =
-    checkpoints
-      .find(_.name == checkpointName) match {
-      case Some(checkpointConfig) =>
-        runCheckpoint(checkpointConfig)
-      case None =>
-        throw new RestonomerContextException(s"The checkpoint: $checkpointName does not exists.")
-    }
+  def runCheckpoint(checkpointName: String): Unit = {
+    val filePath = s"$CHECKPOINTS_CONFIG_DIRECTORY_PATH/$checkpointName.conf"
```

Comment:
This assumption is wrong that any checkpoint file will be placed just immediate to checkpoint folder. We can also have subdirectories within checkpoints folder and have our checkpoint file within any of the subdirectory.
You need to find the file with the corresponding name and its path in the complete checkpoints hierarchy using tail recursion.

Reply:
You can take reference from loadConfigsFromDirectory() method in RestonomerConfigurationsLoader class.

Reply:
resolved